src/luarocks/fs/lua.lua

```diff
@@ -544,86 +544,183 @@ end
 if socket_ok then
 
 local ltn12 = require("ltn12")
+local http = require("socket.http")
 local luasec_ok, https = pcall(require, "ssl.https")
 
-local redirect_protocols = {
-   http = http,
-   https = luasec_ok and https,
-}
+-- Patch ltn12 source used by LuaSocket to get fixed length
```

Comment:
There are a few large rocks (1MiB or more), like sailor and luacheck (the fancy documentation fonts are probably worth it), for them this progress line is really nice. But I agree that the connection setup is typically the longest part. Perhaps combining progress line with a spinner is optimal, and this PR should be a work in progress for now.